apps/api/src/domain/models/term/repositories/arango-query-repository-provider.ts

```diff
@@ -28,6 +30,10 @@ export class ArangoQueryRepositoryProvider implements IQueryRepositoryProvider {
             return this.audioItemQueryRepository;
         }
 
+        if (resourceType === ResourceType.photograph) {
```

Comment:
I don't want to get carried away with this pattern. On the one hand, we could dynamically register query repositories. Beyond that, we might want to question why we use the provider pattern. I'm guessing it might be useful for generic event handlers, i.e., event handlers that work for multiple or all resource types. In that case, it is definitely worth keeping as it makes it easy to swap  out all our Arango implementations for a different persistence mechanism (other db \ in memory) as a single set (abstract factory \ kit).